ntbs-service/DataAccess/AlertRepository.cs

```diff
@@ -70,6 +71,15 @@ public async Task<IList<UnmatchedLabResultAlert>> GetAllOpenUnmatchedLabResultAl
                 .ToListAsync();
         }
 
+        public async Task<DataQualityDraftAlert> GetOpenDraftAlertForNotificationAsync(int notificationId)
+        {
+            return (DataQualityDraftAlert) await _context.Alert
+                .Where(n => n.AlertStatus != AlertStatus.Closed)
+                .Where(n => n.NotificationId == notificationId)
+                .Where(n => n.AlertType == AlertType.DataQualityDraft)

```

Comment:
I think an `OfType<DataQualityDraftAlert>()` will plumb into EF's inheritance mapping better (and remove the need for the cast).